app/controllers/foods_controller.rb

```diff
@@ -1,2 +1,34 @@
 class FoodsController < ApplicationController
+  def index
+    @foods = Food.all
+  end
+
+  def new
+    @user = current_user
+    @food = Food.new
+  end
+
+  def create
+    @food = Food.new(food_params)
+    if @food.save
+      flash[:notice] = 'Food created successfully.'
+      redirect_to foods_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @food = Food.find(params[:id])
+    authorize! :destroy, @food
+    @food.destroy
+    flash[:success] = 'Food is removed successfully'
+    redirect_to root_path
```

Comment:
**[Optional]** I suggest redirecting to the food list `redirect_to foods_path` rather than the root.